### DISCUSSION TITLE
Black profile setting

### DISCUSSION THREAD

**Discussion by steakhutzeee** [microsoft/vscode-isort#144] [Q&A] [answered]
Hi,

from the documentation i see that the extension can be configured with:

`"isort.args":["--profile", "black"],`

Doing this i see the following in VSCode:

![isort](https://user-images.githubusercontent.com/57228102/200172548-042c9bcf-34df-4df9-aa9c-9b3804271db1.JPG)

Is this correct or in VSCode i should put all the string on the same line? Is it ok that the setting is split on 2 lines in VSCode UI?

Thank you in advance!

**Comment by karthiknadig** [comment, marked as answer]
That is OK. 

The important thing is that the setting in JSON file looks like this:
```
"isort.args": ["--profile", "black"],
```

**Reply by steakhutzeee** [reply]
Thank you!

Little question. The comma is needed even if this one is the last line in the line, just before the " } " ?

**Reply by karthiknadig** [reply]
No, you don't need the comma, if it is the last line in the JSON:
```json
{
    "isort.check": true,
    "isort.args": ["--profile", "black"]
}
```

**Reply by steakhutzeee** [reply]
I see, but at this point it's not clear what `"isort.check": true` does.

**Reply by karthiknadig** [reply]
`"isort.check": true` enables `isort` to run with `--check` and detect if there are any import sort issues. If there are any issue it will add Hints to  quick fix that issue.

**Reply by steakhutzeee** [reply]
I added it and looks like isort works fine like before, i do not see any message. I would expect --check to not change the file but just point hints. Correct?

**Reply by karthiknadig** [reply]
Yes you should see gray dots below the first import:
![image](https://user-images.githubusercontent.com/3840081/201428919-ca838217-06fb-477c-baf5-425c979a4bba.png)

You can change the level to warning, and it will show up in problems window:
```json
{
    "isort.check": true,
    "isort.severity": {
        "W": "Warning",
        "E": "Warning"
    },
    "isort.args": ["--profile", "black"]
}
```
![image](https://user-images.githubusercontent.com/3840081/201429083-28611222-cd72-4806-94de-b847bcd4cfa9.png)

**Reply by steakhutzeee** [reply]
Strange, this is not working for me. The errors are just automatically fixed by isort. Like --check is ignored.

Here is the isort section from my settings.json:

```
"isort.check": true,
"isort.args": [
       "--profile",
       "black"
],
"isort.severity": {
      "E": "Warning
}
```

Maybe because i have:

```
"[python]": {
    "editor.codeActionsOnSave": {
           "source.organizeImports": true
    },
}
```

?

**Reply by karthiknadig** [reply]
If you have `"source.organizeImports": true` then `imports` gets fixed, so `--check` won't fail.

**Reply by steakhutzeee** [reply]
Yes, confirming this. Thanks!
